hardware/raspberrypi/spi/README.md

```diff
@@ -3,51 +3,91 @@
-
-### Shell
-
-```bash
-# Write binary 1, 2 and 3
-echo -ne "\x01\x02\x03" > /dev/spidev0.0
-```
+The Raspberry Pi family of devices is equipped with three [SPI](https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus) buses. The main SPI interface is referred to as SPI0 in the documentation; the second is SPI1. SPI2 is only usable on a Compute Module or Compute Module 3. Devices based on BCM2711, i.e. Pi 4B and the upcoming Compute Module 4, have access to an additional 4 SPI buses (SPI3 to SPI6).
 
 <a name="hardware"></a>
 ## Hardware
 
-The BCM2835 on the Raspberry Pi has 3 SPI Controllers. Only the SPI0 controller is available on the header.
-Chapter 10 in the [BCM2835 ARM Peripherals](../bcm2835/BCM2835-ARM-Peripherals.pdf) datasheet describes this controller.
+The BCM2835 on the Raspberry Pi has 3 [http://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus SPI] Controllers. The main SPI bus (with two hardware chip selects) is available on the header of all Pis with Linux kernel support. The second SPI bus (with three hardware chip selects) is available on 40-pin versions of Pis, with kernel support from Raspbian Jessie 2016-05-10 distribution and up. The third SPI bus (also with three hardware chip selects) is only usable on a Compute Module because the pins aren't brought out onto the 40-pin header.
+
+BCM2711 adds another 4 SPI buses, each with 2 hardware chip selects. All are available on the 40-pin header (provided nothing else is trying to use the same pins).
+
+Chapter 10 in the [http://www.raspberrypi.org/wp-content/uploads/2012/02/BCM2835-ARM-Peripherals.pdf BCM2835 ARM Peripherals] datasheet describes the main controller.  Chapter 2.3 describes the auxiliary controller.
+
+### Pin/GPIO mappings
+
+#### SPI0 (available on J8/P1 headers on all RPi versions)
+| SPI Function | Header Pin | Broadcom Pin Name | Broadcom Pin Name |
+|---|---|---|---|
+| MOSI | 19 | GPIO10 | SPI0_MOSI |
+| MISO | 21 | GPIO09 | SPI0_MISO |
+| SCLK | 23 | GPIO11 | SPI0_SCLK |
+| CE0  | 24 | GPIO08 | SPI0_CE0_N |
+| CE1  | 26 | GPIO07 | SPI0_CE1_N |
+
+#### SPI1 (available only on 40-pin J8 header)
+| SPI Function | Header Pin | Broadcom Pin Name | Broadcom Pin Name |
+|---|---|---|---|
+| MOSI | 38 | GPIO20 | SPI1_MOSI |
+| MISO | 35 | GPIO19 | SPI1_MISO |
+| SCLK | 40 | GPIO21 | SPI1_SCLK |
+| CE0  | 12 | GPIO18 | SPI1_CE0_N |
+| CE1  | 11 | GPIO17 | SPI1_CE1_N |
+| CE2  | 36 | GPIO16 | SPI1_CE2_N |
+
+#### SPI2 (available only on Compute Modules)
+| SPI Function | Broadcom Pin Name | Broadcom Pin Name |
+|---|---|---|
+| MOSI | GPIO41 | SPI2_MOSI |
+| MISO | GPIO40 | SPI2_MISO |
+| SCLK | GPIO42 | SPI2_SCLK |
+| CE0  | GPIO43 | SPI2_CE0_N |
+| CE1  | GPIO44 | SPI2_CE1_N |
+| CE2  | GPIO45 | SPI2_CE2_N |
+
+#### SPI3 (BCM2711 only)
+| SPI Function | Header Pin | Broadcom Pin Name | Broadcom Pin Name |
+|---|---|---|---|
+| MOSI | 03 | GPIO02 | SPI3_MOSI |
+| MISO | 28 | GPIO01 | SPI3_MISO |
```

Comment:
Is is worth adding a small note that this conflicts with the HAT EEPROM pins?

Reply:
Be my guest.